### ISSUE TITLE
handleErr returns non-original Error object with loss of details 

### ISSUE THREAD

**Issue by ewrayjohnson** [rlwhatley3/awaits#3]
Love this utility!

The handleErr message "instanceof Error" can obfuscate the original Error.
The problem is most visible in Jest unit testing when using a custom environment.
For more info see: https://github.com/facebook/jest/issues/2549#issuecomment-783580711

Here is the problem code:
`	return err instanceof Error ? [err, null] : [new Error(err), null];`

The following code fixes the issue:
`
    return typeof err === 'object' &&
        err.constructor.prototype.name === Error.prototype.name? 
        [err, null] : [new Error(err), null];
`

**Comment by rlwhatley3** [comment]
Awesome, thank you for this, I'll try to implement and test the above and push out some changes.

**Comment by rlwhatley3** [comment]
@ewrayjohnson can you provide an example of usage that is generating this issue for you.

**Comment by rlwhatley3** [comment]
@ewrayjohnson I've updated a few things, namely, I loosed the check on what an 'error' is through some basic duck typing. Now it checks an OR statement to see if the 'error' implements  'message' and 'stack', which is kind of the minimums for what I consider to be an error.  

I think that using the prototype name comparison as you suggested could generate additional issues if something like a TypeError, or RangeError, or really anything other than a generic error were to ever get rejected directly (the name would never match).

I've also added a new method that implements allSettled if you want to take a look at that. I published version 10.0.0, which has all the latest changes.
